omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,199 @@
+		executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+	}
+	
+	@Override
+	public String getURI() {
+		return "idgen/autogenerationoption";
+	}
+	
+	@Override
+	public String getUuid() {
+		return AUTO_GENERATION_OPTION_UUID;
+	}
+	
+	private int getId() {
+		return Integer.parseInt(getUuid());
+	}
+	
+	@Override
+	public long getAllCount() {
+		Integer allAutoGenerationOptions = 0;
+		for (PatientIdentifierType patientIdentifierType : Context.getPatientService().getAllPatientIdentifierTypes()) {
+			allAutoGenerationOptions += identifierSourceService.getAutoGenerationOptions(patientIdentifierType).size();
+		}
+		return allAutoGenerationOptions;
+	}
+	
+	@Test()
+	public void shouldCreateNewAutoGenerationOption() throws Exception {
+		long originalCount = getAllCount();
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("location", LOCATION);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", true);
+		autoGenerationOption.add("manualEntryEnabled", true);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		assertNotNull( PropertyUtils.getProperty(newAutoGenerationOption, "source"));
+		assertNotNull( PropertyUtils.getProperty(newAutoGenerationOption, "location"));
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "identifierType"));
+		assertEquals(true, PropertyUtils.getProperty(newAutoGenerationOption, "automaticGenerationEnabled"));
+		assertEquals(true, PropertyUtils.getProperty(newAutoGenerationOption, "manualEntryEnabled"));
+	}
+	
+	@Test()
+	public void shouldCreateNewAutoGenerationOptionWithoutLocation() throws Exception {
+		long originalCount = getAllCount();
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", true);
+		autoGenerationOption.add("manualEntryEnabled", true);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		assertNotNull( PropertyUtils.getProperty(newAutoGenerationOption, "source"));
+		assertNotNull( PropertyUtils.getProperty(newAutoGenerationOption, "location"));
```

Comment:
Instead of assertNotNull, wouldn't assertEquals be more precise? Because it may not be null, but have a value different from the one you expect. Not so?